sklearn/utils/validation.py

```diff
@@ -903,17 +896,24 @@ def check_is_fitted(estimator, attributes, msg=None, all_or_any=all):
     NotFittedError
         If the attributes are not found.
     """
+    if attributes != 'deprecated':
+        warnings.warn("Passing attributes to check_is_fitted is deprecated"
+                      "and will be removed in 0.23. The attributes "
+                      "argument is ignored.", DeprecationWarning)
+    if isclass(estimator):
+        raise TypeError("{} is a class, not an instance.".format(estimator))
     if msg is None:
         msg = ("This %(name)s instance is not fitted yet. Call 'fit' with "
                "appropriate arguments before using this method.")
 
     if not hasattr(estimator, 'fit'):
         raise TypeError("%s is not an estimator instance." % (estimator))
 
-    if not isinstance(attributes, (list, tuple)):
```

Comment:
something like:

```python
 from ..pipeline import Pipeline
    if isinstance(estimator, Pipeline):
        for est in estimator:
            if est is not None:
                check_is_fitted(est)
    else:
        attrs = [v for v in vars(estimator)
                 if (v.endswith("_") or v.startswith("_"))
                 and not v.startswith("__")]

        if not attrs:
            raise NotFittedError(msg % {'name': type(estimator).__name__})
```

Reply:
we could... but the old version was also not applicable to pipeline, and arguably that's an issue with the pipeline API modifying its init arguments, right?

Reply:
True we should an attribute `estimators_` we should exist only when calling `fit`.

I will have to write something custom for `partial_dependence`. I am unsure if this is better to have something in `partial_depedence` )and potentially other places where we want to check that a pipeline is fitted) or modify `check_is_fitted` and fix it when `Pipeline` are fixed.

Reply:
I'm fine with putting it in this function, but I would argue it's orthogonal to this PR.

Reply:
OK fair enough.